src/web/signer.ts

```diff
@@ -126,174 +113,116 @@ export class TurboWebArweaveSigner extends TurboDataItemAbstractSigner {
   }
 }
 
-export async function streamSigner(
-  {
-    streamFactory,
-    signer,
-    fileSize,
-    emitter,
-  }: {
-    streamFactory: () => ReadableStream<Uint8Array>;
-    signer: Signer;
-    fileSize: number;
-    emitter?: TurboEventEmitter;
+export const readableStreamToAsyncIterable = (
+  stream: ReadableStream<Uint8Array>,
+): AsyncIterable<Buffer> => ({
```

Comment:
hmm, should this return `AsyncIterable<Uint8Array`?

Reply:
Sure - technically the same thing though. But for clarity/consistency can go with the typed array